### ISSUE TITLE
Problems with API docs, specifically for native pages

### ISSUE THREAD

**Issue by Sfshaza** [flutter/flutter#18702]
**tl;dr**
- Search doesn't work for native docs.
- Native docs are very hard to find. (Right now, the `PLATFORM_INTEGRATI...` entries are several pages down (buried) in the side nav.)
- The native content is not great.
- Native docs look very different than our Dart docs.

---

Ben K gave me a demonstration of how hard (impossible) it is to find docs for the native packages on [docs.flutter.io](https://docs.flutter.io/). This matters because he's about to land a feature that allows devs to write plugins that support isolates running in the background, even when the app is closed. It's difficult to write a plugin when you can't search the native docs, unless you know exactly what you're looking for.

1. Try searching in the search textfield for, say, `BinaryMessenger`. Zero hits. How about `FlutterViewController`. Nope.
2. Next, try:

```
BinaryMessenger site:docs.flutter.io  
```

Now you can see that the class page is down in [`docs.flutter.io/javadoc/io/flutter/plugin/common/BinaryMessenger.html`](https://docs.flutter.io/javadoc/io/flutter/plugin/common/BinaryMessenger.html).

3. Now try searching in the browser for 

```
FlutterViewController site:docs.flutter.io
```

The page is located in [`docs.flutter.io/objcdoc/Classes/FlutterViewController.html`](https://docs.flutter.io/objcdoc/Classes/FlutterViewController.html)

4. Next, go back to the API docs home page by clicking "Flutter Docs" in the upper left corner. Nope, you're still in the objective C directory. Once you've entered [`docs.flutter/javadoc`](https://docs.flutter.io/javadoc/), you're in the same boat. There is no way to get back to the regular Flutter docs from either native set of docs, without manually modifying the URL.

---

In the short term (pre 1.0), I suggest that we:
* Expose buttons on the front page that direct readers to the iOS and Android docs.
* Add buttons to the native pages that take you back to the regular docs.
* Welcome PRs to improve the docs.

In the long term (post 1.0), this site needs more love and care. Can we assign someone to improve it?

@bkonyi, @Hixie

**Comment by Hixie** [comment]
We should probably split this into separate bugs.

cc @cbracken @jason-simmons @chinmaygarde for content creation
cc @gspencergoog @jcollins-g for dartdoc integration

What we really need to do here is hire one or two people to implement javadoc and objcdoc tooling that will integrate properly with dartdoc. cc @eseidel for that problem.

**Comment by matthew-carroll** [comment]
@Hixie @Sfshaza Has anyone started on the Javadoc side of this?  I've been writing Javadocs for the embedding refactor.  Hopefully that work isn't being superseded elsewhere?

**Comment by Hixie** [comment]
@matthew-carroll In our experience with framework docs, (a) nobody is ever working on the same docs as anyone else, and (b) if that ever happens, you get even better docs (because the two versions tend to not really be redundant with each other, if they're each detailed enough, and you get to just merge the best parts of each). So I wouldn't worry about anyone else also writing docs.

**Comment by pedromassangocode** [comment]
I was able to reproduce this on the current API docs website.
